webxrlayers-1.bs

```diff
@@ -221,15 +224,15 @@ enum XRLayerLayout {
  - A layout of <dfn enum-value for="XRLayerLayout">stereo-left-right</dfn> indicates that the layer is in stereo and divided left to right.
  - A layout of <dfn enum-value for="XRLayerLayout">stereo-top-bottom</dfn> indicates that the layer is in stereo and divided top to bottom.
 
- NOTE: If an {{XRLayer}} is created with an {{XRLayerLayout/"stereo"}} layout, it is highly recommended that it is allocated with an {{XRTextureType/"texture-array"}} texture type.
+ NOTE: If an {{XRGenericLayer}} is created with an {{XRLayerLayout/"stereo"}} layout, it is highly recommended that it is allocated with an {{XRTextureType/"texture-array"}} texture type.
 
-XRLayer {#xrlayertype}
--------
-Every layer except XRWebGLLayer is derived from XRLayer.
-XRLayer defines a set of common attributes and behaviors acress all layer types.
+XRGenericLayer {#xrgenericbaselayertype}
+--------------
+Every layer except XRWebGLLayer is derived from XRGenericLayer.
+XRGenericLayer defines a set of common attributes and behaviors acress all layer types.
 
 <pre class="idl">
-interface XRLayer {
+interface XRGenericLayer {
```

Comment:
I don't like this name, but that's a very minor concern.

Reply:
One other option, I guess, is that since the amount of shared values here is limited and there's certain attributes that we're sharing across all but one layer type already, maybe we just skip having this intermediate layer type all together and have each layer copy these into their own interface.

It's not the most elegant thing ever, but there's not a whole lot that this structure concretely buys us anyway besides preventing you from passing an `XRWebGLLayer` into `get[View]SubImage()`. But even that will need to make sure that the layer was not created by a different `XR____Binding` object in it's algorithm anyway (because we don't want WebGL bindings to return images for WebGPU-created layers.) So maybe i wouldn't be a terrible things to drop this type from the hierarchy.

Reply:
> If we end up keeping `XRGenericLayer` it's fine, it just sounds slightly awkward to me.

I like the name `XRCompositionLayer`. I will update the PR with that name.

Reply:
> One other option, I guess, is that since the amount of shared values here is limited and there's certain attributes that we're sharing across all but one layer type already, maybe we just skip having this intermediate layer type all together and have each layer copy these into their own interface.

There's still quite a bit of common interface between the composition layers so I think we should keep a common base class.